src/hash/rpo/mod.rs

```diff
@@ -116,18 +112,15 @@ impl Hasher for Rpo256 {
         let mut i = 0;
         let mut buf = [0_u8; 8];
         for chunk in bytes.chunks(BINARY_CHUNK_SIZE) {
-            if i < num_elements - 1 {
-                buf[..BINARY_CHUNK_SIZE].copy_from_slice(chunk);
-            } else {
-                // if we are dealing with the last chunk, it may be smaller than BINARY_CHUNK_SIZE
-                // bytes long, so we need to handle it slightly differently. We also append a byte
-                // with value 1 to the end of the string; this pads the string in such a way that
-                // adding trailing zeros results in different hash
-                let chunk_len = chunk.len();
-                buf = [0_u8; 8];
-                buf[..chunk_len].copy_from_slice(chunk);
-                buf[chunk_len] = 1;
+            // if we are dealing with the last chunk, it may be smaller than BINARY_CHUNK_SIZE
+            // bytes long, so we need to handle it slightly differently. We also append a byte
+            // with value 1 to the end of the string; this pads the string in such a way that
+            // adding trailing zeros results in different hash
+            if chunk.len() < BINARY_CHUNK_SIZE {
+                buf.fill(0);
+                buf[chunk.len()] = 1;
             }
+            buf[..BINARY_CHUNK_SIZE.min(chunk.len())].copy_from_slice(chunk);
```

Comment:
nit: I'd probably break up this code into two branches. Maybe something like:
```Rust
if chunk.len() < BINARY_CHUNK_SIZE {
    buf.fill(0);
    buf[chunk.len()] = 1;
    buf[..chunk.len()].copy_from_slice(chunk);
} else {
    buf[..BINARY_CHUNK_SIZE].copy_from_slice(chunk);
}
```
I think this is a bit more readable.